src/components/application_manager/include/application_manager/rpc_protection_mediator.h

```diff
@@ -0,0 +1,96 @@
+ * this list of conditions and the following
+ * disclaimer in the documentation and/or other materials provided with the
+ * distribution.
+ *
+ * Neither the name of the Ford Motor Company nor the names of its contributors
+ * may be used to endorse or promote products derived from this software
+ * without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#ifndef SRC_COMPONENTS_APPLICATION_MANAGER_INCLUDE_PROTOCOL_HANDLER_RPC_PROTECTION_MEDIATOR_H_
+#define SRC_COMPONENTS_APPLICATION_MANAGER_INCLUDE_PROTOCOL_HANDLER_RPC_PROTECTION_MEDIATOR_H_
+
+#include <cstdint>
+#include <memory>
+
+namespace ns_smart_device_link {
+namespace ns_smart_objects {
+class SmartObject;
+}  // namespace ns_smart_objects
+}  // namespace ns_smart_device_link
+namespace smart_objects = ns_smart_device_link::ns_smart_objects;
+
+namespace application_manager {
+class Application;
+}  // namespace application_manager
+
+namespace application_manager {
+/*
+* \brief RPCProtectionMediator interface
+*/
+class RPCProtectionMediator {
+ public:
+  /*!
+   * \brief virtual destructor RPCProtectionMediator
+   */
+  virtual ~RPCProtectionMediator() {}
+
+  /*
+   * \param function_id function id
+   * \param app smart pointer to Application
+   * \param conrrelation_id conrrelation id
+   * \param is_rpc_service_secure the flag the secure service started
+   * \return true if function need encryption for current app,  else false
+   */
+  virtual bool DoesRPCNeedEncryption(const uint32_t function_id,
+                                     std::shared_ptr<Application> app,
+                                     const uint32_t conrrelation_id,
+                                     const bool is_rpc_service_secure) = 0;
+  /*
+   * \param conrrelation_id conrrelation id
+   * \return true if the message with correlation id correlation_id needed e
+   * ncryption else false
+   */
+  virtual bool DoesRPCNeedEncryption(const uint32_t conrrelation_id) = 0;
+  /*
+  * \brief massage will be encrypted by force
+  * \param conrrelation_id conrrelation id
+  */
+  virtual void EncryptByForce(const uint32_t conrrelation_id) = 0;
```

Comment:
@BSolonenko i think more extensive description is needed here, since the use cases of this function are not obvious

Reply:
@mked-luxoft  extensive description added